api/src/main/java/org/apache/iceberg/types/Types.java

```diff
@@ -451,6 +452,10 @@ public VariantType asVariantType() {
       return this;
     }
 
+    Object writeReplace() throws ObjectStreamException {
+      return new PrimitiveHolder(toString());
```

Comment:
That's a great addition, but since `Variant` does not inherit from `PrimitiveType`, I'm not sure if it will cause ambiguity with the name and definition of the `PrimitiveHolder` class.

Reply:
Yeah. That's why I didn't make the change initially. Variant type works more like primitive types in Iceberg schema level and `PrimitiveHolder` is only for internal use. Maybe I can update the comment to make it clear as follows:

> /** Replacement for primitive types or Variant in Java Serialization. */

Reply:
Make sense